core-plugins/src/test/java/co/cask/hydrator/plugin/batch/sink/KinesisSinkTest.java

```diff
@@ -84,6 +86,6 @@ public void test() throws Exception {
     // Throws illegal argument exception because of invalid AWS keys
     mrManager.start();
     mrManager.waitForFinish(5, TimeUnit.MINUTES);
-
+    Assert.assertEquals(mrManager.getHistory().get(0).getStatus(), ProgramRunStatus.FAILED);
```

Comment:
just as a sanity check, assert that there's only 1 record in history first.
Otherwise, you might get a NoSuchElementException, if list size is 0, and you try to access the 0th index.